source/opt/upgrade_memory_model.cpp

```diff
@@ -68,6 +69,22 @@ void UpgradeMemoryModel::UpgradeInstructions() {
   // instructions. Additionally, Workgroup storage class variables and function
   // parameters are implicitly coherent in GLSL450.
 
+  // Upgrade modf and frexp first since they generate new stores.
+  for (auto& func : *get_module()) {
+    func.ForEachInst([this](Instruction* inst) {
+      if (inst->opcode() == SpvOpExtInst) {
+        auto import =
+            get_def_use_mgr()->GetDef(inst->GetSingleWordInOperand(0u));
+        if (reinterpret_cast<char*>(import->GetInOperand(0u).words.data()) ==
+            std::string("GLSL.std.450")) {
+          auto ext_inst = inst->GetSingleWordInOperand(1u);
+          if (ext_inst == GLSLstd450Modf || ext_inst == GLSLstd450Frexp) {
```

Comment:
For speed reasons, I'd first test against ext_inst being Modf or Frexp.  Then do the string compare test. (The ext_inst opcode is always present)

Reply:
Done